renderdoc/driver/vulkan/vk_core.cpp

```diff
@@ -703,6 +703,75 @@ void WrappedVulkan::SetDebugMessageSink(WrappedVulkan::ScopedDebugMessageSink *s
   Threading::SetTLSValue(debugMessageSinkTLSSlot, (void *)sink);
 }
 
+void WrappedVulkan::MarkPendingCommandBufferAsDeleted(VkCommandBuffer commandBuffer)
+{
+  SCOPED_LOCK(m_PendingCmdBufferCallbacksLock);
+
+  for(PendingCommandBufferCallbacks &pending : m_PendingCmdBufferCallbacks)
+  {
+    if(pending.commandBuffer == commandBuffer)
+    {
+      pending.commandBuffer = VK_NULL_HANDLE;
+      return;
+    }
+  }
+}
+
+void WrappedVulkan::AddPendingCommandBufferCallbacks(VkCommandBuffer commandBuffer)
+{
+  SCOPED_LOCK(m_PendingCmdBufferCallbacksLock);
```

Comment:
This lock should be grabbed right before `m_PendingCmdBufferCallbacks.push_back` to minimise contention so that it's not held while calling into the driver.